model/event/event.go

```diff
@@ -15,41 +16,43 @@ const (
 	TaskLogCollection = "task_event_log"
 )
 
-type Event interface {
-	ID() string
-
-	// return the r_type bson field
-	Type() string
-
-	// Time returns the time that the event occurred
-	Time() time.Time
-
-	// Processed is whether or not this event has been processed. An event
-	// which has been processed has successfully have notifications intents
-	// created and stored, but does not indicate whether or not these
-	// notifications have been successfully sent to all recipients
-	// If true, the time is the time that this event was marked as
-	// processed. If false, time is the zero time
-	Processed() (bool, time.Time)
-
-	// MarkProcessed marks this event as processed at the current
-	// time
-	MarkProcessed() error
-
-	// Notifications fetches all subscriptions relevant to this event.
-	// Events may have no subscribers.
-	// TODO: specify return type in EVG-2861
-	Notifications() ([]interface{}, error)
-}
-
 type EventLogEntry struct {
+	ID           bson.ObjectId `bson:"_id" json:"-"`
+	ResourceType string        `bson:"r_type,omitempty" json:"resource_type,omitempty"`
+	ProcessedAt  time.Time     `bson:"processed_at,omitempty" json:"processed_at,omitempty"`
+
 	Timestamp  time.Time `bson:"ts" json:"timestamp"`
 	ResourceId string    `bson:"r_id" json:"resource_id"`
 	EventType  string    `bson:"e_type" json:"event_type"`
 	Data       Data      `bson:"data" json:"data"`
 }
 
+// Type returns the resource type (i.e. event data type). It will reflect into
+// the Data interface if required.
+func (e *EventLogEntry) Type() string {
+	if len(e.ResourceType) == 0 {
+		_, rtype := findResourceTypeIn(e.Data)
+		return rtype
```

Comment:
Should this set `e.ResourceType` before returning?

Reply:
This change is adding compatibility with new format, which we'll let bake for a bit. Then EVG-2951 will do that (and bake for a bit). Then, EVG-2952 will remove the code for the old resource type, and we'll run the migration.